cmd/tsconnect/build-pkg.go

```diff
@@ -41,4 +47,33 @@ func runBuildPkg() {
 		log.Fatalf("Type generation failed: %v", err)
 	}
 
+	if err := updateVersion(); err != nil {
+		log.Fatalf("Cannot update version: %v", err)
+	}
+
+	log.Printf("Built package version %s", version.Long)
+}
+
+func updateVersion() error {
+	packageJSONBytes, err := os.ReadFile("pkg-package.json")
+	if err != nil {
+		return fmt.Errorf("Could not read package.json: %w", err)
+	}
+
+	var packageJSON map[string]any
+	packageJSONBytes, err = hujson.Standardize(packageJSONBytes)
+	if err != nil {
+		return fmt.Errorf("Could not standardize template package.json: %w", err)
+	}
+	if err := json.Unmarshal(packageJSONBytes, &packageJSON); err != nil {
+		return fmt.Errorf("Could not unmarshal package.json: %w", err)
+	}
+	packageJSON["version"] = version.Long
```

Comment:
Looks like the version info that gets embedded looks like this `1.29.0-tf23741275`. I'm curious how package managers like npm handle hashes at the end like that… is there a reason we don't want to truncate down to a semver string?

Reply:
They're OK with it -- it gets treated as a [prerelease version identifier in semver](https://semver.org/#spec-item-9). This is generally good, because we don't bump the patch version in OSS, and without the Git hash it we would always end up with `1.29.0` and thus no way to distinguish.